h2o-persist-hdfs/src/main/java/water/persist/PersistHdfs.java

```diff
@@ -552,4 +551,14 @@ public boolean delete(String path) {
       throw new HDFSIOException(path, CONF.toString(), e);
     }
   }
+
+  @Override
+  public boolean canHandle(String path) {
+    URI uri = new Path(path).toUri();
+    try {
+      return FileSystem.getFileSystemClass(uri.getScheme(), CONF) != null;
```

Comment:
It would probably be better to handle the case of unspecified scheme (when uri.getScheme() == null). If someone defined an "null" filesystem "fs.null.impl", he would see unexpected behavior.